### PR TITLE
Testing Patch

### PR DESCRIPTION
**Pull request by djsd123**
This is a blueprint for the functional tests.  I have included tests for
our NFS solution

### CODE REVIEW

File: spec/nfs/ec2_spec.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'colorize'
+
+tag_name       = 'tag:Name'
+tag_val        = NAME
```
Comment (xoen) [inline spec/nfs/ec2_spec.rb]: Who sets `NAME`?

File: spec/nfs/ec2_spec.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  filters: [
+    {
+      name: tag_name.to_s,
+      values: [
+        tag_val.to_s
+      ]
+    }
+  ]
+).each do |inst|
+  ec2_instances.store(inst.id, inst.security_groups)
```
Comment (xoen) [inline spec/nfs/ec2_spec.rb]: Never heard of `Hash#store()` before, what's the difference with doing `ec2_instances[inst.id] = inst.security_groups`?

File: spec/nfs/ec2_spec.rb (hunk 1 above)
Comment (djsd123) [reply spec/nfs/ec2_spec.rb]: That's set as a constant in `spec_helper.rb` line [8](https://github.com/ministryofjustice/analytical-platform-infrastructure/pull/5/files#diff-93830fa29d616f7c87903d08b5b1b29aR8)

Comment (djsd123) [reply spec/nfs/ec2_spec.rb]: Thinking about it 🤔 .  probably shouldn't.

**Comment by djsd123** [comment]
I've somehow mangled this PR with missing files. Going to fix.

**Comment by djsd123** [comment]
sorted 👍

File: spec/nfs/ec2_spec.rb (hunk 1 above)
Comment (djsd123) [reply spec/nfs/ec2_spec.rb]: Moved var.

File: spec/nfs/ec2_spec.rb (hunk 2 above)
Comment (djsd123) [reply spec/nfs/ec2_spec.rb]: 🤷‍♂ tbh.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+require 'awspec'
+require 'aws-sdk-iam'
+require 'aws-sdk-ec2'
+
+REGION = 'eu-west-1'
+
+creds = Aws::SharedCredentials.new(profile_name: 'data')
```
Comment (djsd123) [inline spec/spec_helper.rb]: @sh31 may want to modify this when he creates the piepline

**Pull request merged**